### PR TITLE
fix: better tests

### PR DESCRIPTION
**Pull request by AdityaHarvi**
#### Summary of changes/features added:
- Proper intent tests make books, run tests, and then delete the book leaving a clean slate.
- Also removed the cancel button from the camera screen.

#### Screenshots of UI changes (update with each new commit): 
- Just removed the cancel button, thats basically it ;-;

## Pull Request Checklist
- [ ] Unit test written?
- [x] (**For later in project**) Integration tests written?
- [x] Java docs written?

### CODE REVIEW

File: client/app/src/androidTest/java/com/example/criengine/AddABookTest.java [hunk 1]
```diff
@@ -58,19 +60,61 @@ public void setUp() {
...
+
+        assertTrue(solo.waitForText("Before you go...", 1, 1000));
+        solo.clickOnText("NOT NOW");
+
+        solo.assertCurrentActivity("Wrong Activity", RootActivity.class);
+
+        solo.clickInList(0);
+        assertTrue(solo.waitForText("Mock Title", 1, 1000));
+        solo.goBack();
+        cleanup();
```
Comment (rjmah1) [inline client/app/src/androidTest/java/com/example/criengine/AddABookTest.java:91]: isn't this what @After is for?

File: client/app/src/androidTest/java/com/example/criengine/AddABookTest.java [hunk 2]
```diff
@@ -82,6 +126,18 @@ public void scanButtonTest() {
         solo.assertCurrentActivity("Wrong Activity", ScanActivity.class);
     }
 
+    /**
+     * Delete the book to make it ready for the next test.
+     */
+    public void cleanup() {
```
Comment (rjmah1) [inline client/app/src/androidTest/java/com/example/criengine/AddABookTest.java:132]: make this @After for tests

File: client/app/src/androidTest/java/com/example/criengine/CameraScreenTest.java [hunk 3]
```diff
@@ -143,23 +136,44 @@ public void checkConfirmationDialog() {
...
+        solo.enterText((EditText) solo.getView(R.id.newBookAuthor), "This is a new Author");
+        solo.enterText((EditText) solo.getView(R.id.newBookISBN), "This is a new ISBN");
+        solo.clickOnButton("Save and Add Photo");
+        solo.clickOnText("NOT NOW");
+    }
+
+    /**
+     * Delete the book to make it ready for the next test.
+     */
+    public void cleanup() {
```
Comment (rjmah1) [inline client/app/src/androidTest/java/com/example/criengine/CameraScreenTest.java:163]: similarly @After

File: client/app/src/androidTest/java/com/example/criengine/ViewMyBookTest.java [hunk 4]
```diff
@@ -107,22 +100,48 @@ public void cancelEditTitleTest() {
...
         solo.assertCurrentActivity("Wrong Activity", MyBookActivity.class);
+
+        solo.goBack();
+        cleanup();
+    }
+
+    /**
+     * Add a mock book.
+     */
+    public void addBook() {
```
Comment (rjmah1) [inline client/app/src/androidTest/java/com/example/criengine/ViewMyBookTest.java:118]: If these are all the same, you can make it into a static method of another class that does this for all the cases, similarly for cleanup (especially if it's not before and after all the tests)

**Review by rjmah1 — CHANGES_REQUESTED** [review]

File: client/app/src/androidTest/java/com/example/criengine/AddABookTest.java (hunk 1 above)
Comment (AdityaHarvi) [reply client/app/src/androidTest/java/com/example/criengine/AddABookTest.java:91]: not every test has a book to be removed so it shouldn't go into the after

File: client/app/src/androidTest/java/com/example/criengine/ViewMyBookTest.java (hunk 4 above)
Comment (AdityaHarvi) [reply client/app/src/androidTest/java/com/example/criengine/ViewMyBookTest.java:118]: Sure can do

File: client/app/src/androidTest/java/com/example/criengine/CameraScreenTest.java (hunk 3 above)
Comment (AdityaHarvi) [reply client/app/src/androidTest/java/com/example/criengine/CameraScreenTest.java:163]: I can do this 👍

**Review by shadowciaw — APPROVED** [review]

**Review by rjmah1 — APPROVED** [review]
lgtm

**Pull request merged**